Setup GitHub Pages Deployment Pipeline for Keycloak Documentation

Marc-Jr opened the pull request:
This pull request implements a complete pipeline for deploying Keycloak operations documentation to GitHub Pages. It addresses the ticket: "Setup a pipeline for deploying Keycloak CC docs on GitHub Pages." 
Changes include:
MkDocs setup: Added mkdocs.yml with Material theme and structured docs (index.md, architecture.md, documentation.md).
GitHub Actions: Workflow in .github/workflows/deploy-docs.yml automates building and deploying the site to GitHub Pages with Python setup and caching.
Supporting files: Added requirements.txt, updated .gitignore to exclude site/ and Python envs, and updated README.md.

closes #4

Marc-Jr committed:
Initial MkDocs setup with Material theme

- Created Python virtual environment for dependency isolation
- Installed mkdocs 1.6.1 and mkdocs-material 9.6.21
- Configured mkdocs.yml with Material theme and navigation
- Created documentation structure: getting-started, installation, operations, reference
- Added initial content to 11 documentation pages
- Verified local development server works correctly

Resolves: #4 (partial - documentation setup complete, CI/CD pipeline next)

Marc-Jr committed:
Add GitHub Actions workflow for documentation deployment

- Created workflow to build and deploy MkDocs documentation
- Triggers on push to main and pipeline-deployment branches
- Manual workflow dispatch enabled
- Uses ubuntu-latest with Python 3.x
- Builds with MkDocs Material theme
- Deploys to GitHub Pages using official actions
- Configured proper permissions and concurrency

Part of Issue #4

Marc-Jr committed:
Add requirements.txt for GitHub Actions cache

- Added Python dependencies file for pip caching
- Fixes GitHub Actions workflow cache error

Part of Issue #4

Marc-Jr committed:
Add comprehensive README.md with project overview, setup instructions, and documentation guidelines

Marc-Jr committed:
Merge branch 'main' of https://github.com/ADORSYS-GIS/keycloak-ops into pipeline-deployment

Marc-Jr committed:
Simplify documentation to 2 pages per supervisor feedback

- Reduced from 11 pages to 2 pages (Home, Documentation)
- Removed getting-started, installation, operations, and reference sections
- Simplified mkdocs.yml navigation configuration
- Created minimal README with essential information only
- Streamlined content for initial documentation phase
- Maintained professional Material theme and core functionality

Changes:
- Deleted: docs/getting-started/ (3 files)
- Deleted: docs/installation/ (2 files)
- Deleted: docs/operations/ (3 files)
- Deleted: docs/reference/ (2 files)
- Updated: docs/index.md (simplified)
- Created: docs/documentation.md (single documentation page)
- Updated: mkdocs.yml (minimal navigation)
- Created: README.md (minimal project info)

Result: Clean, minimal documentation site with 2 pages as requested

Marc-Jr committed:
Remove architecture.md from documentation

Guy-Ghis commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,54 @@
+    branches:
+      - main
+      - pipeline-deployment  # Also deploy from your branch for testing
+  workflow_dispatch:  # Allow manual triggering
```
- Allow the workflow to be triggered only by a push to the main branch
- Disable workflow dispatch too

Marc-Jr replied:
Got it, I’ll update the workflow so it only runs on pushes to the main branch. Thanks for catching that.

Guy-Ghis commented on README.md:
```diff
@@ -1 +1,27 @@
+
+## Live Site
+
+[View Documentation](https://marc-jr.github.io/keycloak-ops/)
```
Modify this link to be that of the main repo and not your fork

Marc-Jr committed:
Updated deploy-doc.yml and README.md

Marc-Jr committed:
Update workflow to trigger only on docs changes

- Removed workflow_dispatch (manual trigger)
- Added paths filter to trigger only on:
  - docs/** (any documentation file changes)
  - mkdocs.yml (configuration changes)
  - requirements.txt (dependency changes)
- Prevents unnecessary builds when non-docs files change
- More efficient CI/CD pipeline per supervisor feedback

Guy-Ghis reviewed (approved):
Thanks for assessing the requested changes.

Guy-Ghis reviewed (approved):
Good.

Guy-Ghis reviewed (dismissed):
Good job.

Nkwenti-Severian-Ndongtsop commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,56 @@
+
+      - name: Install dependencies
+        run: |
+          pip install mkdocs mkdocs-material
```
@Marc-Jr i noticed you are installing just mkdocs mkdocs-material
it would have been good if you install the dependencies directly from the requirements.txt file as you did in the README.md file
so in the future in case of adding dependencies, they will be no need modifying the workflow file

Nkwenti-Severian-Ndongtsop commented on .github/workflows/deploy-docs.yml:
```diff
@@ -0,0 +1,56 @@
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
```
@Marc-Jr for the python version, i think it will be better you choose a version and use it for the pipeline so that it should not be falling to the latest, which in the future might not be compatible with what we are doing completely
 what do you think ??

Marc-Jr committed:
Update workflow per code review feedback

Changes based on reviewer suggestions:
- Changed Python version from '3.x' to '3.11' for stability
- Updated dependency installation to use requirements.txt
- Ensures version consistency across builds
- Improves maintainability and prevents future compatibility issues

Addresses review comments from @Nkwenti-Severian-Ndongtso

Marc-Jr committed:
Add manual trigger and pipeline-deployment branch support

- Added workflow_dispatch for manual triggering
- Added pipeline-deployment to trigger branches
- Keeps path filters for efficient builds
- Python 3.11 pinned for stability
- Uses requirements.txt for dependencies

Marc-Jr committed:
Add changelog page to trigger workflow

- Created changelog.md documenting version 1.0.0
- Updated navigation to include changelog
- This commit modifies docs/ to trigger automated deployment

Marc-Jr committed:
Finalize workflow and remove test changelog

- Reverted workflow to trigger only on main branch
- Removed workflow_dispatch (no manual trigger needed)
- Removed pipeline-deployment from trigger branches
- Deleted changelog.md (was added for testing)
- Updated mkdocs.yml back to 2 pages (Home, Documentation)
- Production-ready configuration per code review requirements

marcjazz reviewed (changes requested):
Please provide a guide on how to update the GitHub pages docs website.

Marc-Jr committed:
Add contributing guidelines for documentation

- Created contributing.md with step-by-step guide
- Explains how to add new documentation pages
- Includes best practices for writing docs
- Covers local testing and git workflow
- Provides directory structure reference
- Simplified for current 2-page documentation setup
- Helps team members contribute effectively

Marc-Jr committed:
Modified docs/contributing.md
